### PR TITLE
fix: validate UTF-8 boundaries on mutation byte ranges

### LINKED ISSUE #141: fix: validate UTF-8 boundaries on mutation byte ranges
`src/mutator.rs:50-68` — mutations use byte ranges with no validation that they fall on UTF-8 character boundaries. A range cutting through a multi-byte character will panic or produce corrupted output on non-ASCII source files.

### PR DESCRIPTION
**Pull request by Darkroom4364**
## Summary
- Add `is_char_boundary` checks in `mutation_diff` before string slicing to prevent panics on multibyte characters
- Validate `range.start <= range.end` in runner before splice (catch inverted ranges)
- Validate byte range bounds in mutator before extracting original text
- Add tests for multibyte chars and invalid boundary detection

Closes #141

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved validation to prevent crashes when processing mutations with invalid ranges or special characters.
  * Stricter boundary checks during mutation application to catch invalid positions early.

* **Tests**
  * Extended test coverage for edge cases involving special character handling.

### CODE REVIEW

File: src/mutator.rs [hunk 1]
```diff
@@ -108,6 +108,12 @@ pub fn generate_mutations(
                         continue;
                     }
 
+                    if candidate.byte_range.start > candidate.byte_range.end
+                        || candidate.byte_range.end > source.len()
+                    {
+                        continue;
+                    }
```
[the remark this answers is not in the document]
Comment (Darkroom4364) [reply src/mutator.rs:115]: Not needed. All byte ranges come from tree-sitter's node.byte_range(), node.start_byte(), and node.end_byte(), which always align to token boundaries in the parsed source. Tree-sitter parses UTF-8 source and produces ranges that respect character boundaries by design.

Adding is_char_boundary checks here would be a no-op guard against something that can't happen with the current operator implementations. The from_utf8_lossy on line 120 is already a safe fallback for the theoretical case, and the is_char_boundary guard we added in mutation_diff (report/mod.rs) protects the actual panic-prone string slicing path.
